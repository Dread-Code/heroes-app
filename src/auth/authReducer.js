import { LOGIN, LOGOUT } from "../types/types";

export const authReducer = (state = {}, action) => {
  switch (action?.type) {
    case LOGIN:
      return {
        ...action.payload,
        logged: true,
      };
    case LOGOUT:
      return {
        logged: false,
      };

    default:
      return state;
  }
};
