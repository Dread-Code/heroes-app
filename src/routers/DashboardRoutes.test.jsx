import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { DashboardRoutes } from "./DashboardRoutes";
import { AuthContext } from "../auth/AuthContext";

describe("DashboardRoutes", () => {
  const contextValue = {
    user: { logged: true, name: "Lucas" },
    dispatch: jest.fn(),
  };
  test("should march snapshoot", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
