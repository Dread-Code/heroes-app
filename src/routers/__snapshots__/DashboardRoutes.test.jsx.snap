// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRoutes should march snapshoot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Lucas
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-5"
  >
    <div>
      <h1>
        Marvelscreen
      </h1>
      <hr />
      <div
        className="animate__animated animate__fadeIn row row-cols-1 row-cols-md-3 g-4"
      >
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-spider"
              onClick={[Function]}
            >
              <img
                alt="Spider Man"
                className="card-img"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Spider Man
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Peter Parker
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  Amazing Fantasy #15
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-captain"
              onClick={[Function]}
            >
              <img
                alt="Captain America"
                className="card-img"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Captain America
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Steve Rogers
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  Captain America Comics #1
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-iron"
              onClick={[Function]}
            >
              <img
                alt="Iron Man"
                className="card-img"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Iron Man
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Tony Stark
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  Tales of Suspense #39
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-thor"
              onClick={[Function]}
            >
              <img
                alt="Thor"
                className="card-img"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Thor
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Thor Odinson
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  Journey into Myster #83
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-hulk"
              onClick={[Function]}
            >
              <img
                alt="Hulk"
                className="card-img"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hulk
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Bruce Banner
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  The Incredible Hulk #1
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-wolverine"
              onClick={[Function]}
            >
              <img
                alt="Wolverine"
                className="card-img"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wolverine
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                James Howlett
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  The Incredible Hulk #180
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-daredevil"
              onClick={[Function]}
            >
              <img
                alt="Daredevil"
                className="card-img"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Daredevil
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Matthew Michael Murdock
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  Daredevil #1
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-hawkeye"
              onClick={[Function]}
            >
              <img
                alt="Hawkeye"
                className="card-img"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hawkeye
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Clinton Francis Barton
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  Tales of Suspense #57
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-cyclops"
              onClick={[Function]}
            >
              <img
                alt="Cyclops"
                className="card-img"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Cyclops
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Scott Summers
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  X-Men #1
                </small>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <a
              href="/hero/marvel-silver"
              onClick={[Function]}
            >
              <img
                alt="Silver Surfer"
                className="card-img"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </a>
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Silver Surfer
              </h5>
              <p
                className="card-text"
              >
                <strong>
                  Alter ego: 
                </strong>
                 
                Norrin Radd
              </p>
              <p
                className="card-text"
              >
                <strong
                  style={
                    Object {
                      "fontSize": "2em",
                    }
                  }
                >
                  Marvel Comics
                </strong>
                <br />
                <small
                  className="text-muted"
                >
                  The Fantastic Four #48
                </small>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
